tooling/generators/provider/provider.tmpl.js

```diff
@@ -9,7 +10,11 @@ import {Http} from 'angular2/http';
 */
 @Injectable()
 export class <%= jsClassName %> {
-  constructor(@Inject(Http) http) {
+  static get parameters(){
+    return [[Http]]
+  }  
+
+  constructor(http) {
```

Comment:
Can remove the this.data and this.http since TypeScript will add them with `data: any = null` and the constructor param.